feat: add tags and enhance searching

gera2ld opened the pull request:
Editing tags:
![image](https://github.com/violentmonkey/violentmonkey/assets/3139113/e6404e8f-ffab-4c6a-bb6a-a82fe8efb0d8)
Display of tags:
![image](https://github.com/violentmonkey/violentmonkey/assets/3139113/4a23eaba-2c32-4a97-ae1f-e2bbc8595b5f)
Search:
![image](https://github.com/violentmonkey/violentmonkey/assets/3139113/a2a5bf2e-75b4-48be-8a3a-fc88fbe754f7)

tophf commented:
Awesome!

But the hint is too big to be shown by default and it is overlaid by the built-in autocomplete dropdown. I suggest adding an (i) icon that shows this info on hover and maybe also on click (via showMessage) for mobile users.

![image](https://github.com/violentmonkey/violentmonkey/assets/1310400/173c24f3-fcde-4d9e-945a-3b07d1e8eda4)

The input is too wide. The previous width is sufficient + the new `field-sizing` CSS or a polyfill for older browsers to autosize the field on typing.

The config icon after the input is not vertically centered:
![image](https://github.com/violentmonkey/violentmonkey/assets/1310400/07cc6806-d132-48f1-902f-cee1981537b8)

The sort order selector is hidden again, but I revealed it intentionally because I think it's just as important as search.

The tooltip is bugged at the edge:
![chrome_2024-01-22_18-43-52](https://github.com/violentmonkey/violentmonkey/assets/1310400/5ab74dcc-7345-45cf-91b8-952bd2b95767)

`/regexp/` notation was removed, but this form is widely used in software e.g. in devtools filters, while `re:` prefix is unusual.

gera2ld commented:
> The input is too wide. The previous width is sufficient + the new field-sizing CSS or a polyfill for older browsers to autosize the field on typing.

I thought it could be wider to contain more combined conditions, but yeah the previous width seems okay. BTW the previous layout overlaps with the refresh button in small screens. I'll see if I can find a better way to define the width.

> `/regexp/` notation was removed, but this form is widely used in software e.g. in devtools filters, while `re:` prefix is unusual and it makes searching for regexp in code impossible or weird like `code:re:foo`

The reason I chose `code+re:foo` over `code:/foo/` is to avoid tedious escaping, e.g. `code+re:https://www.google.com/` vs `code:/https:\/\/www.google.com\//`. After a second thought, I guess we can search with plain text in such cases, so it should not be a problem.

tophf commented:
There's no need to escape `/` inside regexps currently, i.e. `/\.com/foo/bar//` works correctly and it should remain so.

gera2ld commented:
Then how to tell if `/foo/ /bar/` is one or two regular expressions?

tophf commented:
An embedded space is an unsupported case, same as in devtools, and the common workaround is `\s`.

tophf commented:
Just checked our current implementation and it allows spaces because it only supports regexp as the entire search input value.

gera2ld commented:
Updated UI.
Also added support for simple regexp without spaces: `/foobar/ code:/foobar/`, but still kept `code+re:"foo bar"` to support regexps with spaces.

tophf commented on src/options/utils/search.js:
```diff
@@ -0,0 +1,90 @@
+          negative,
+        });
+      } else {
+        const reMatches = pattern.match(/^\/(.*?)\/(\w*)$/);
```
I think this should be the first thing to try, otherwise it won't match when the entire input is `/foo #bar/`

gera2ld replied:
This is intended. I don't want to match regexp with spaces this way, as it is ambiguous and confusing.
So `/foo #bar/` will be parsed into string `/foo` and tag `bar/`. To input a regexp we can use `re:"foo #bar"` or `/foo\s#bar/`.

tophf replied:
Strictly speaking the new behavior is correct, but it's not really useful because tags aren't likely to end in `/`...

gera2ld replied:
Yeah, by right they should be two strings since `/` is not allowed in tags. But an invalid tag matches nothing so it does not hurt. I think it might be a bad idea for `#...` to be sometimes a tag and sometimes not.

tophf commented on src/_locales/en/messages.yml:
```diff
@@ -914,6 +920,21 @@ titleSearchHint:
+    * Search by script name: <code>name:"awesome name"</code>
+    * Search by script code: <code>code:"awesome code"</code>
+    * Negative search: <code>!#tag2 !name:"unwanted"</code>
+    * Regular expressions: <code>name:/awesome_\w+/ name+re:"awesome \w+ with space"</code>
```
I'd like to separate them visually and restore the old examples:
```html
<code>/\w+?/</code>, <code>/\w+?/gi</code>, <code>name:/\w+?/</code>, <code>name+re:"\w+? with space"</code>
```
